### PR TITLE
WIP: Updates: feedback 2023-07-29 (PART 1)

### PR DESCRIPTION
**Pull request by clarklindev**
- using jwt tokens for authorization and authentication
- remove http-only cookies
- remove frontend (will create a branch)

- clean up auth controller code
  - unit test for authorization middleware (isAuth) 
  - refactor auth controller (fixed error handling flow)
  - formatted api returns to be JSONAPI format
- clean up contact controller code
- unit test custom validators
- unit tests Auth Controller
- unit tests Contact Controller

TODO:

### CODE REVIEW

File: backend/src/apis/contacts/helpers/findContact.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import Contact from '../../../lib/models/contact';
+import { IContact } from '../../../lib/interfaces/IContact';
+
+//function receives object with query attributes...eg. {email:"abc@gmail.com"}
```
Comment (captainmike) [inline backend/src/apis/contacts/helpers/findContact.ts:4]: Not necessary for all your methods. Where applicable, I would suggest using JSDoc format. vscode would also understand it.

File: backend/src/index.ts [hunk 2]
```diff
@@ -1,64 +1,59 @@
...
+import { IError } from './lib/interfaces/IError';
+import DateHelper from './lib/helpers/DateHelper';
+import { jsonApiErrorResponseFromError } from './lib/helpers/jsonApiErrorResponseFromError';
 
 dotenv.config();
 
 const app: Express = express();
-
-//mongodb atlas node 2.2.12
 const MONGODB_URI = `mongodb://${process.env.MONGODB_USER}:${process.env.MONGODB_PASSWORD}@ac-yztvzc4-shard-00-00.b5tvnqi.mongodb.net:27017,ac-yztvzc4-shard-00-01.b5tvnqi.mongodb.net:27017,ac-yztvzc4-shard-00-02.b5tvnqi.mongodb.net:27017/?ssl=true&replicaSet=atlas-dbcw9j-shard-0&authSource=admin&retryWrites=true&w=majority`;
```
Comment (captainmike) [inline backend/src/index.ts:17]: If `MONGODB_USER` and `MONGODB_USER` are always used in `MONGODB_URI`, I would suggest using `MONGODB_URI` directly as the environment variable. Currently I'm not seeing how you can point to different environments either; which is obviously a big consideration for an _environment_ variable.

File: backend/src/index.ts [hunk 3]
```diff
@@ -1,64 +1,59 @@
...
   res.status(404).json({ status: 'ERROR', message: 'Page Not Found' });
 });
 
 // catches all errors passed with next(err);
-app.use((err: ErrorWithStatus, req: Request, res: Response, next: NextFunction) => {
-  console.log(err);
-  const status = err.statusCode || 500;
-  const message = err.message;
-  res.status(status).json({ message });
+//note you need "next:NextFunction" prop as its part of the middleware error handler function prop signature
```
Comment (captainmike) [inline backend/src/index.ts:54]: I would suggest introducing the convention `_` for unused parameters.

**Review by captainmike — APPROVED** [review]
Looks decent. Sees suggestions. Then let's get this in.

File: backend/src/index.ts (hunk 3 above)
Comment (clarklindev) [reply backend/src/index.ts]: done

File: backend/src/index.ts [hunk 4]
```diff
@@ -1,64 +1,59 @@
...
-// Enable CORS for specific origins with credentials
 const corsOptions = {
   origin: `${process.env.FRONTEND_URL}:${process.env.FRONTEND_PORT}`, // Replace with your frontend URL
-  credentials: true,
 };
 
-//order important needs to come before express.json()
-app.use(cors(corsOptions));
+//cors order important: needs to come before express.json()
+// app.use(cors(corsOptions));
```
[the remark this answers is not in the document]
Comment (clarklindev) [reply backend/src/index.ts]: done

File: backend/src/index.ts (hunk 2 above)
Comment (clarklindev) [reply backend/src/index.ts]: done. combined MONGODB_USER and MONGODB_PASSWORD as MONGODB_URI

**Pull request closed without merging**